### **Phase 1: Business Understanding**

#### **1.1 Business Context**

Small and Medium Enterprises (SMEs) are vital contributors to Ghana’s economy ,they create jobs, stimulate innovation, and contribute significantly to GDP.

However, banks that finance these SMEs often face challenges such as:

- Irregular repayment patterns  
- Sector-specific risk exposure  
- Limited data-driven insights to guide lending decisions  

As a result, financial institutions struggle to balance profitability with risk management in SME lending portfolios.

This project aims to provide data-driven insights into the performance of SME loans, helping banks optimize lending strategies, minimize defaults, and enhance financial stability.


#### **1.2 Problem Statement**

Despite SMEs’ importance, banks frequently experience inefficient loan allocation and high default rates due to inadequate analysis of sector behavior and repayment performance.

 


#### **1.3 Project Objectives**

The project seeks to:

- Identify repayment trends across different SME sectors in Ghana.
- Calculate and visualize key portfolio performance metrics:
  1. **Default rates**
  2. **Average loan sizes**
  3. **Repayment periods**
- Provide actionable insights to help banks optimize SME lending strategies, reduce credit risk, and enhance financial inclusion.


#### **1.4 Analytical Goals**

To achieve the objectives, I will:

- Determine repayment and default behavior patterns
- Quantify sector-level and portfolio-level performance
- Identify relationships between interest rates, loan sizes, and repayment outcomes


#### **1.5 Success Criteria**

##### **Business Success Criteria**
- The analysis should clearly identify which sectors and business sizes have the highest default risks.
- Insights should guide loan officers toward data-driven SME lending strategies.

##### **Data Mining Success Criteria**
- Accurate computation of key KPIs (default rate, repayment rate, loan distribution, etc.).
- Interactive dashboards and reports using Power BI to visualize trends and comparisons.


#### **1.6 Key Stakeholders**

| **Stakeholder**         | **Role**              | **Interest**                                                                                 |
|-------------------------|-----------------------|-----------------------------------------------------------------------------------------------|
| Bank Management         | Decision-makers       | Want to understand risk exposure and optimize SME loan allocation.                          |
| Credit Risk Analysts    | Data interpreters     | Need risk segmentation insights for each sector.                                              |
| Loan Officers           | Operational users     | Use findings to improve lending and repayment monitoring.                                     |



#### **1.7 Constraints and Assumptions**

##### **Constraints**
- Missing repayment data for ongoing loans may limit accuracy.
- Dataset covers only 530 records — may not represent the entire SME market.
- No external macroeconomic data (e.g., inflation, interest rate trends).

##### **Assumptions**
- `Repayment_Status` accurately reflects repayment behavior.
- `Due_Date` and `Issue_Date` represent actual contractual timelines.
- Risk ratings (`Sector_Risk_Rating`) are pre-determined by bank policies.





### **Phase 2: Data Understanding**

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [3]:
#load dataset
sme_loan_data = pd.read_csv('../Data/SME_loan_data.csv', index_col=0)


sme_loan_data.head()


SME_Name              Sector      Region  Loan_Amount (GHC)  \
Loan_ID                                                                
L0001    AGR_0001         Agriculture     Central             447224   
L0002    AGR_0002         Agriculture     Western             329270   
L0003    TRA_0003               Trade  Upper East             401715   
L0004    ENE_0004  Energy & Utilities    Northern             424502   
L0005    FIS_0005           Fisheries  Upper West             214431   

         Interest_Rate Issue_Date    Due_Date Repayment_Status Business_Size  \
Loan_ID                                                                        
L0001            17.40  9/12/2020   6/11/2023        Defaulted         Small   
L0002            11.64  7/23/2023   9/19/2024             Paid        Medium   
L0003            28.71  6/21/2019  10/26/2020        Defaulted        Medium   
L0004            17.70   1/2/2020   4/10/2022     Late Payment         Micro   
L0005            22.17  7/21/2021    1/8/2023     Late Payment        Medium   

         Collateral_Value (GHC)  Repayment_Amount (GHC)  Loan_Term (Months)  \
Loan_ID                                                                       
L0001                     89202               525040.98                33.0   
L0002                    321741               367597.03                14.0   
L0003                    264392               517047.38                16.0   
L0004                    157430               499638.85                27.0   
L0005                    128747               261970.35                17.0   

         Amount_Paid (GHC)  Date_Paid  Amount_Due  Credit_Score  Risk_Rating  
Loan_ID                                                                       
L0001             27326.28  5/12/2023   497714.70           NaN          NaN  
L0002            367597.03  9/17/2024        0.00           NaN          NaN  
L0003             27448.39  9/26/2020   489598.99           NaN          NaN  
L0004            499638.85  4/16/2022        0.00           NaN          NaN  
L0005            261970.35  1/10/2023        0.00           NaN          NaN

In [4]:
#data inspection
print("Number of rows and columns in the dataset:", sme_loan_data.shape)

sme_loan_data.info()

Number of rows and columns in the dataset: (513, 17)
<class 'pandas.core.frame.DataFrame'>
Index: 513 entries, L0001 to L0530
Data columns (total 17 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   SME_Name                513 non-null    object 
 1   Sector                  513 non-null    object 
 2   Region                  513 non-null    object 
 3   Loan_Amount (GHC)       513 non-null    int64  
 4   Interest_Rate           513 non-null    float64
 5   Issue_Date              513 non-null    object 
 6   Due_Date                513 non-null    object 
 7   Repayment_Status        513 non-null    object 
 8   Business_Size           513 non-null    object 
 9   Collateral_Value (GHC)  513 non-null    int64  
 10  Repayment_Amount (GHC)  509 non-null    float64
 11  Loan_Term (Months)      510 non-null    float64
 12  Amount_Paid (GHC)       513 non-null    float64
 13  Date_Paid               513 non-null    o

In [5]:
#converting date type for date columns from object to datetime
sme_loan_data['Issue_Date'] = pd.to_datetime(sme_loan_data['Issue_Date'])
sme_loan_data['Due_Date'] = pd.to_datetime(sme_loan_data['Due_Date'])
sme_loan_data['Date_Paid'] = pd.to_datetime(sme_loan_data['Date_Paid'])
sme_loan_data['Risk_Rating'] = sme_loan_data['Risk_Rating'].astype('object')


sme_loan_data.dtypes



SME_Name                          object
Sector                            object
Region                            object
Loan_Amount (GHC)                  int64
Interest_Rate                    float64
Issue_Date                datetime64[ns]
Due_Date                  datetime64[ns]
Repayment_Status                  object
Business_Size                     object
Collateral_Value (GHC)             int64
Repayment_Amount (GHC)           float64
Loan_Term (Months)               float64
Amount_Paid (GHC)                float64
Date_Paid                 datetime64[ns]
Amount_Due                       float64
Credit_Score                     float64
Risk_Rating                       object
dtype: object

In [6]:
#checking for missing values
sme_loan_data.isnull().sum()

SME_Name                    0
Sector                      0
Region                      0
Loan_Amount (GHC)           0
Interest_Rate               0
Issue_Date                  0
Due_Date                    0
Repayment_Status            0
Business_Size               0
Collateral_Value (GHC)      0
Repayment_Amount (GHC)      4
Loan_Term (Months)          3
Amount_Paid (GHC)           0
Date_Paid                   0
Amount_Due                  5
Credit_Score              513
Risk_Rating               513
dtype: int64

In [7]:
# Calculate missing value count and percentage
missing_summary = sme_loan_data.isnull().sum().to_frame(name='Missing_Count')
missing_summary['Missing_Percentage'] = (sme_loan_data.isnull().mean() * 100).round(2)

missing_summary

missing_summary.sort_values(by='Missing_Percentage', ascending=False)


Missing_Count  Missing_Percentage
Risk_Rating                       513              100.00
Credit_Score                      513              100.00
Amount_Due                          5                0.97
Repayment_Amount (GHC)              4                0.78
Loan_Term (Months)                  3                0.58
SME_Name                            0                0.00
Sector                              0                0.00
Loan_Amount (GHC)                   0                0.00
Region                              0                0.00
Business_Size                       0                0.00
Repayment_Status                    0                0.00
Due_Date                            0                0.00
Issue_Date                          0                0.00
Interest_Rate                       0                0.00
Collateral_Value (GHC)              0                0.00
Amount_Paid (GHC)                   0                0.00
Date_Paid                           0                0.00

###  Data Preparation – Missing Value Treatment

After profiling the dataset for missing values, the following results were obtained:

| Column                     | Missing_Count | Missing_Percentage |
|---------------------------|----------------|--------------------|
| Risk_Rating               | 513            | 100.00             |
| Credit_Score              | 513            | 100.00             |
| Amount_Due                | 5              | 0.97               |
| Repayment_Amount (GHC)    | 4              | 0.78               |
| Loan_Term (Months)        | 3              | 0.58               |
| SME_Name                  | 0              | 0.00               |
| Sector                    | 0              | 0.00               |
| Loan_Amount (GHC)         | 0              | 0.00               |
| Region                    | 0              | 0.00               |
| Business_Size             | 0              | 0.00               |
| Repayment_Status          | 0              | 0.00               |
| Due_Date                  | 0              | 0.00               |
| Issue_Date                | 0              | 0.00               |
| Interest_Rate             | 0              | 0.00               |
| Collateral_Value (GHC)    | 0              | 0.00               |
| Amount_Paid (GHC)         | 0              | 0.00               |
| Date_Paid                 | 0              | 0.00               |



#### ✅ Missing Value Handling Strategy

| Missing Percentage Range | Action Taken |
|--------------------------|--------------|
| **Less than 5%**         | Conditional Imputation |
| **Between 5% and 20%**   | Conditional Imputation |
| **Greater than 20%**     | Derived based on business logic |



#### ✅ Decisions Based on This Dataset

##### 1. Columns with < 5% Missing Values

- **Repayment_Amount (GHC): 0.78%**
  - Formula used:  
    ```
    Repayment Amount = Loan Amount + (Loan Amount × (Interest Rate / 100))
    ```

- **Loan_Term (Months): 0.58%**
  - Formula used:  
    ```
    Loan Term = Difference in months between Issue_Date and Due_Date
    ```

- **Amount_Due: 0.97%**
  - Formula used:  
    ```
    Amount Due = Repayment Amount – Amount Paid
    ```

**Action:** Conditional imputation — no rows dropped.



##### 2. Columns with 100% Missing Values  
These missing fields were derived based on business logic.

###### Credit Score

Credit Score components:

| Component                   | Score Range |
|-----------------------------|-------------|
| Collateral Coverage Score   | 0 – 40      |
| Business Size Score         | 10 – 30     |
| Repayment Behaviour Score   | 0 – 30      |

**Formulas Used**

1. **Collateral Score**
Collateral Coverage = (Collateral Value / Loan Amount) × 40
(maximum allowed score = 40)


2. **Business Size Score**

| Business Size | Score |
|---------------|--------|
| Micro         | 10     |
| Small         | 20     |
| Medium        | 30     |

3. **Repayment Behaviour Score**

| Repayment Status       | Score |
|------------------------|--------|
| Paid                  | 30     |
| Late Payment          | 15     |
| Defaulted             | 0      |

✅ Final Credit Score:
Credit Score = Collateral Score + Business Size Score + Repayment Score



#####  Risk Rating

Assigned based on the computed **Credit Score**:

| Credit Score Range | Risk Rating |
|--------------------|-------------|
| 0 – 50             | High Risk   |
| 51 – 70            | Medium Risk |
| 71 – 100           | Low Risk    |

